pyairtable/utils.py

```diff
@@ -10,7 +10,9 @@ def datetime_to_iso_str(value: datetime) -> str:
     Args:
         value: datetime object
     """
-    return value.isoformat(timespec="milliseconds") + "Z"
+    if value.utcoffset() is None:
+        return value.isoformat(timespec="milliseconds") + "Z"
```

Comment:
Woudln't it make more sense to add a timezone and use the same isoformat function below for all calls? Also is it correct to just add UTC to a datetime object that is not timezone aware? I guess that can end in weird and unwanted behavior.

Reply:
Thinking about this some more time I think we should rather fail here, if not timezone aware object was passed in. Also it seems not correctly implemented in the first place. But I might be wrong, as I've not tested the code, so please feel free to give me some more feedback. Implementing timezones is quite fuzzy, you have to be concentrated.

Reply:
So I made some tests myself, as I also needed this functionality. You have two options to pass to this function:
a) a timezone aware object
b) a timezone unaware object (naive)

In both cases, the syntax `value.isoformat(timespec="milliseconds")` will work. The `+Z` was not required in first place it seems. Airtable will treat naive datetime strings as UTC.

Personally I'd implement it that naive datetime objects should be added with the local system timezone:
If called without arguments (or with tz=None) the system local timezone is assumed for the target timezone. The .tzinfo attribute of the converted datetime instance will be set to an instance of [timezone](https://docs.python.org/3/library/datetime.html#datetime.timezone) with the zone name and offset obtained from the OS.

Reply:
Thanks for your review! Just one comment and I'd happy to commit the revisions.